debezium-core/src/main/java/io/debezium/transforms/TimezoneConverter.java

```diff
@@ -0,0 +1,502 @@
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withValidation(Field::isListOfRegex)
+            .withDescription("A comma-separated list of rules that specify what events should be evaluated for timezone conversion, using one of the following formats: "
+                    + "`source:<tablename>`:: Matches only Debezium change events with a source information block with the specified table name. All time-based fields will be converted. "
+                    + "`source:<tablename>:<fieldname>`:: Matches only Debezium change events with a source information with the specified table name. Only the specified field name will be converted. "
+                    + "`topic:<topicname>`:: Matches the specified topic name, converting all time-based fields. "
+                    + "`topic:<topicname>:<fieldname>`:: Matches the specified topic name, converting only the specified field name. "
+                    + "`<matchname>:<fieldname>`:: Uses a heuristic matching algorithm to matches the source information block table name if the source information block exists, otherwise matches against the topic name. The conversion is applied only to to the specified field name. ");
+
+    private static final Field EXCLUDE_LIST = Field.create("exclude.list")
+            .withDisplayName("Exclude List")
+            .withType(ConfigDef.Type.LIST)
+            .withWidth(ConfigDef.Width.MEDIUM)
+            .withImportance(ConfigDef.Importance.HIGH)
+            .withValidation(Field::isListOfRegex)
+            .withDescription("A comma-separated list of rules that specify what events should be excluded from timezone conversion, using one of the following formats: "
+                    + "`source:<tablename>`:: Matches only Debezium change events with a source information block with the specified table name. All time-based fields will be excluded. "
+                    + "`source:<tablename>:<fieldnames>`:: Matches only Debezium change events with a source information with the specified table name. Only the specified field name will be excluded. "
+                    + "`topic:<topicname>`:: Matches the specified topic name, excluding all time-based fields. "
+                    + "`topic:<topicname>:<fieldnames>`:: Matches the specified topic name, excluding only the specified field name. "
+                    + "`<matchname>:<fieldnames>`:: Uses a heuristic matching algorithm to matches the source information block table name if the source information block exists, otherwise matches against the topic name. The conversion is applied only to to the specified field name. ");
+
+    private SmtManager<R> smtManager;
+    private String convertedTimezone;
+    private List<String> includeList;
+    private List<String> excludeList;
+    private static final String SOURCE = "source";
+    private static final String TOPIC = "topic";
+    private static final Pattern TIMEZONE_PATTERN = Pattern.compile("^[+-]\\d{2}:\\d{2}$");
```

Comment:
Please check JavaDoc for `java.time.ZoneOffset.of(String)` to see what are allowed values. From my PoV we can accept simiplified rules where colons are mandatory, hours and minutes are mandatory and there can be optional seconds.